File: TeamCode/src/main/java/org/manchestermachinemakers/RobotConfig.java
package org.manchestermachinemakers;

import com.qualcomm.robotcore.hardware.HardwareMap;

public class RobotConfig {
    public static HardwareMap hwMap;
    public static void activate(HardwareMap hwMap) {
        RobotConfig.hwMap = hwMap;
    }
}

File: TeamCode/src/main/java/org/manchestermachinemakers/RobotConfig.kt
package org.manchestermachinemakers

import com.qualcomm.robotcore.hardware.HardwareMap
import org.manchestermachinemakers.RobotConfig

object RobotConfig {
    var hwMap: HardwareMap? = null
    @JvmStatic
    fun activate(hwMap: HardwareMap) {
        RobotConfig.hwMap = hwMap
    }

    inline fun <reified T> getHardware(name: String) = hwMap?.get(T::class.java, name)
}